x/axelarnet/types/evm_wasm_translator.go

```diff
@@ -0,0 +1,96 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+var (
+	stringType      = funcs.Must(abi.NewType("string", "string", nil))
+	stringArrayType = funcs.Must(abi.NewType("string[]", "string[]", nil))
+	bytesType       = funcs.Must(abi.NewType("bytes", "bytes", nil))
+
+	// abi encoded bytes, with the following format:
+	// wasm method name, argument name list, argument type list, argument value list
+	payloadArguments = abi.Arguments{{Type: stringType}, {Type: stringArrayType}, {Type: stringArrayType}, {Type: bytesType}}
+)
+
+type contractCall struct {
+	SourceChain string `json:"source_chain"`
+	Sender      string `json:"sender"`
+	// Contract is the address of the wasm contract
+	Contract string `json:"contract"`
+	// Msg is a json struct {"methodName": {"arg1": "val1", "arg2": "val2"}}
+	Msg map[string]interface{} `json:"msg"`
+}
+
+// wasm is the json that gets passed to the IBC memo field
+type wasm struct {
+	Wasm contractCall `json:"wasm"`
+}
+
+// ConstructWasmMessage creates a json serialized wasm message from Axelar defined abi encoded payload
+// The abi encoded payload must contain the following information in order
+// - method name (string)
+// - argument names ([]string)
+// - argument types ([]string)
+// - argument values (bytes)
+func ConstructWasmMessage(contractAddr, sourceChain, sender string, payload []byte) ([]byte, error) {
+	args, err := payloadArguments.Unpack(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	methodName := args[0].(string)
+	argNames := args[1].([]string)
+	argTypes := args[2].([]string)
+
+	abiArguments, err := buildArguments(argTypes)
+	if err != nil {
+		return nil, err
+	}
+
+	// unpack to actual argument values
+	argValues, err := abiArguments.Unpack(args[3].([]byte))
```

Comment:
sanity check if this also has the same length (should be the case due to abi decode + above check) but might as well